ProfileManager.js

```diff
@@ -115,6 +123,72 @@ export default class ProfileManager {
     return await this._getAgentContent({profileAgentRecord});
   }
 
+  /**
+   * Gets a capability specified by its reference ID for profile agent from
+   * a cache or delegates one on the fly if it does not exist.
+   *
+   * @param {object} options - The options to use.
+   * @param {string} options.id - The reference ID of the zcap.
+   * @param {string} options.profileAgent - The profile agent requesting a zcap.
+   * @param {boolean} options.useEphemeralSigner - Flag to enable invoking
+   *  capabilities with the ephemeral invocation signer, default `true`.
+   *
+   * @returns {Promise<object>} The capability for the profile agent.
+   */
+  async getAgentCapability({id, profileAgent, useEphemeralSigner = true}) {
+    const capabilityCacheKey = `profileAgent-${profileAgent.id}-zcaps`;
+    const capabilityKey = `${capabilityCacheKey}-${id}-${useEphemeralSigner}`;
+
+    if(!this._cacheContainer.has(capabilityCacheKey)) {
+      this._cacheContainer.set(capabilityCacheKey, new Cache());
+    }
+
+    let promise = this._requests.get(capabilityKey);
+
+    if(promise) {
+      return promise;
+    }
+
+    promise = (async () => {
+      const capabilityCache = this._cacheContainer.get(capabilityCacheKey);
+      let zcap = capabilityCache.get(capabilityKey);
+      if(!zcap) {
+        const agentSigner = await this.getAgentSigner(
+          {profileAgentId: profileAgent.id, useEphemeralSigner: false});
+        const originalZcap = profileAgent.zcaps[id];
+        if(!originalZcap) {
+          const {id: profileAgentId} = profileAgent;
+          throw new Error(
+            `The agent "${profileAgentId}" does not have the zcap: "${id}"`);
+        }
+        if(useEphemeralSigner) {
+          const localSigner = await this.getAgentSigner(
+            {profileAgentId: profileAgent.id, useEphemeralSigner: true});
+          zcap = await utils.delegateCapability({
```

Comment:
Are we tracking anywhere that we want these zcaps to have an expiration ... and that we will need to check that the zcap hasn't expired when we get it from the cache?

Reply:
No, we are not tracking it. Is it the case that we have expiration dates on zcaps? If not, that requires changes deeper in the stack. My suggestion is to make an issue and attack that problem in a separate PR.

Reply:
#35